File: ascenso-backend/src/main/java/com/ais/ascensobackend/ninos/application/services/interfaces/NinoService.java
package com.ais.ascensobackend.ninos.application.services.interfaces;

import com.ais.ascensobackend.ninos.application.dtos.NinoResumen;
import java.time.LocalDate;
import java.util.List;

public interface NinoService {

    List<NinoResumen> listar(Long destacamentoId);

    NinoResumen obtener(Long destacamentoId, Long id);

    NinoResumen crear(
            Long destacamentoId, String nombreCompleto, LocalDate fechaNacimiento, String fotoUrl,
            String encargadoNombre, String encargadoContacto, String contactoEmergencia, LocalDate fechaIngreso,
            Long grupoActualId, Long anioProgramaActualId);

    NinoResumen actualizar(
            Long destacamentoId, Long id, String nombreCompleto, LocalDate fechaNacimiento, String fotoUrl,
            String encargadoNombre, String encargadoContacto, String contactoEmergencia);

    NinoResumen promover(Long destacamentoId, Long id, Long grupoId, Long anioProgramaId);

    void activar(Long destacamentoId, Long id);

    void desactivar(Long destacamentoId, Long id);
}

File: ascenso-backoffice/src/services/ninos/NinoService.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { ActualizarNinoRequest, CrearNinoRequest, Nino, PromoverNinoRequest } from '@/types/ninos'

class NinoService {
  listar(destacamentoId: number) {
    return apiClient.get<Nino[]>(API_ENDPOINTS.ninos.porDestacamento(destacamentoId))
  }

  obtener(destacamentoId: number, id: number) {
    return apiClient.get<Nino>(API_ENDPOINTS.ninos.porId(destacamentoId, id))
  }

  crear(destacamentoId: number, request: CrearNinoRequest) {
    return apiClient.post<Nino>(API_ENDPOINTS.ninos.porDestacamento(destacamentoId), request)
  }

  actualizar(destacamentoId: number, id: number, request: ActualizarNinoRequest) {
    return apiClient.put<Nino>(API_ENDPOINTS.ninos.porId(destacamentoId, id), request)
  }

  promover(destacamentoId: number, id: number, request: PromoverNinoRequest) {
    return apiClient.post<Nino>(API_ENDPOINTS.ninos.promover(destacamentoId, id), request)
  }

  activar(destacamentoId: number, id: number) {
    return apiClient.post<void>(API_ENDPOINTS.ninos.activar(destacamentoId, id))
  }

  desactivar(destacamentoId: number, id: number) {
    return apiClient.post<void>(API_ENDPOINTS.ninos.desactivar(destacamentoId, id))
  }
}

export const ninoService = new NinoService()
